core/src/main/java/org/mule/runtime/core/internal/processor/strategy/ProactorStreamEmitterProcessingStrategyFactory.java

```diff
@@ -174,10 +164,7 @@ protected ReactiveProcessor proactor(ReactiveProcessor processor, ScheduledExecu
         return publisher -> scheduleProcessor(processor, retryScheduler, from(publisher))
             .subscriberContext(ctx -> ctx.put(PROCESSOR_SCHEDULER_CONTEXT_KEY, scheduler));
       } else if (maxConcurrency == MAX_VALUE) {
-        if (processor.getProcessingType() == CPU_INTENSIVE
-            && (processor instanceof InterceptedReactiveProcessor)
-            && SDK_OPERATION_CLASS != null
-            && SDK_OPERATION_CLASS.isAssignableFrom(((InterceptedReactiveProcessor) processor).getProcessor().getClass())) {
+        if (processor instanceof OperationInnerProcessor) {
```

Comment:
this is shady.. this is assuming SDK behavior by testing an interface that any component can implement, not just the SDK. Is this wrong or the comment below should be updated?

Reply:
`OperationInnerProcessor` is not API

Reply:
but still, some other dev which doesn't have the full context could implement that interface and a problem could appear here

Reply:
will add a note on the jdoc for `OperationInnerProcessor `